src/app.ts

```diff
@@ -30,19 +32,22 @@ const scheduleCronJob = async () => {
     fs.writeFileSync(CAROUSEL_DATA_PATH, JSON.stringify(carouselData, null, 2));
   };
 
-  const fetchAndWriteCirculatingSupplyData = async () => {
+  const fetchAndWriteSupplyData = async () => {
     const circulatingSupplyData = await getCirculatingSupply();
+    const totalSupplyData = await getTotalSupply();
 
     fs.writeFileSync(CIRCULATING_SUPPLY_DATA_PATH, JSON.stringify(circulatingSupplyData, null, 2));
+    fs.writeFileSync(TOTAL_SUPPLY_DATA_PATH, JSON.stringify(totalSupplyData, null, 2));
   };
 
   // Fetch data initially such that we have something to serve. There will at most
   // be a buffer of 5 minutes from this running until the first cron execution.
-  if (!fs.existsSync(CIRCULATING_SUPPLY_DATA_PATH)) await fetchAndWriteCirculatingSupplyData();
+  if (!(fs.existsSync(CIRCULATING_SUPPLY_DATA_PATH) && fs.existsSync(TOTAL_SUPPLY_DATA_PATH)))
```

Comment:
Its hard to decode the logic, (maybe I'm just old).
We want to fetch if one of the files is missing, so can we write it this way.
```suggestion
  if (!fs.existsSync(CIRCULATING_SUPPLY_DATA_PATH) || !fs.existsSync(TOTAL_SUPPLY_DATA_PATH))
```

Reply:
I was actually debating what approach to go with here, it probably depends on how you read the logic itself. I'll change it though 💯